src/posix/platform/multithread_api.cpp

```diff
@@ -0,0 +1,260 @@
+            continue;
+        }
+        else
+        {
+            syslog(LOG_INFO, "find empty interface [%s]", iface);
+            break;
+        }
+    }
+
+    return idx;
+}
+
+void *pThreadOtMainLoop(void *arg)
+{
+    pthread_mutex_lock(&gLock);
+    otPlatformConfig platformConfig;
+    int              interfaceIdx                = 0;
+    char             radioUrl[MAX_RADIO_URL_LEN] = {0};
+    char             iface[MAX_INTERFACE_LEN]    = {0};
+    OtConfig        *config                      = (OtConfig *)arg;
+
+    syslog(LOG_INFO, "Inside pThreadOtMainLoop");
+
+    interfaceIdx = getInterface();
+    VerifyOrExit(interfaceIdx != MULTIPLE_INSTANCE_MAX,
+                 syslog(LOG_CRIT, "Interface reached max count...Not able to create new interface"));
+    snprintf(iface, sizeof(iface), "wpan%d", interfaceIdx);
+    syslog(LOG_INFO, "interface found [%s]", iface);
+
+    VerifyOrExit(getRadioURL(config->mComPort) != false, syslog(LOG_CRIT, "radio device not found"));
+    snprintf(radioUrl, sizeof(radioUrl), "spinel+hdlc+uart:///dev/%s", config->mComPort);
+    syslog(LOG_INFO, "radio Url found [%s]", radioUrl);
+    syslog(LOG_INFO, "ot log level [%d]", config->mOtLogLevel);
+
+    platformConfig.mInterfaceName = iface;
+    VerifyOrDie(platformConfig.mRadioUrlNum < OT_ARRAY_LENGTH(platformConfig.mRadioUrls), OT_EXIT_INVALID_ARGUMENTS);
+    platformConfig.mRadioUrls[platformConfig.mRadioUrlNum++] = radioUrl;
+#if defined(__linux)
+    platformConfig.mRealTimeSignal = 41;
+#endif
+    platformConfig.mSpeedUpFactor = 1;
+
+    syslog(LOG_INFO, "Running %s", otGetVersionString());
+    syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
+    IgnoreError(otLoggingSetLevel(config->mOtLogLevel));
+
+    gInstance = otSysInit(&platformConfig);
+    VerifyOrDie(gInstance != NULL, OT_EXIT_FAILURE);
+    syslog(LOG_INFO, "Thread interface: %s", otSysGetThreadNetifName());
+    syslog(LOG_INFO, "RCP version: %s", otPlatRadioGetVersionString(gInstance));
+
+    otLogDebgPlat("ot instance create success!!!");
+
+    pthread_cond_signal(&gCond);
+    pthread_mutex_unlock(&gLock);
+
+    while (gTerminate == false)
+    {
+        otSysMainloopContext mainloop;
+
+        otTaskletsProcess(gInstance);
+
+        FD_ZERO(&mainloop.mReadFdSet);
+        FD_ZERO(&mainloop.mWriteFdSet);
+        FD_ZERO(&mainloop.mErrorFdSet);
+
+        mainloop.mMaxFd           = -1;
+        mainloop.mTimeout.tv_sec  = 10;
+        mainloop.mTimeout.tv_usec = 0;
+
+        otSysMainloopUpdate(gInstance, &mainloop);
+
+        if (otSysMainloopPoll(&mainloop) >= 0)
+        {
+            pthread_mutex_lock(&gLock);
+
+            otSysMainloopProcess(gInstance, &mainloop);
+
+            pthread_mutex_unlock(&gLock);
+        }
+        else if (errno != EINTR)
+        {
+            perror("select");
+            ExitNow();
+        }
+    }
+
+exit:
+    otSysDeinit();
+    gThreadId  = 0;
+    gInstance  = NULL;
+    gTerminate = false;
```

Comment:
In case mutex is not released because of VerifyOrExit, we should send signal and unlock mutex.
So, I think we have to add following lines here. 
```
    pthread_cond_signal(&gCond);
    pthread_mutex_unlock(&gLock);
```